src/ex_cmds2.c

```diff
@@ -181,10 +181,30 @@ dialog_changed(
 	// May get file name, when there is none
 	browse_save_fname(buf);
 #endif
-	if (buf->b_fname != NULL && check_overwrite(&ea, buf,
-				    buf->b_fname, buf->b_ffname, FALSE) == OK)
+	int empty_buf = buf->b_fname == NULL ? TRUE : FALSE;
+	if (empty_buf == TRUE)
+	{
+	    buf->b_fname = vim_strsave((char_u *)"Untitled");
+	    if (STRLEN(NameBuff) == 0 && mch_dirname(NameBuff, MAXPATHL) == OK)
+	    {
+		buf->b_ffname = vim_strsave(NameBuff);
```

Comment:
Now we have the directory in `buf->b_ffname`. But shouldn't this contain the Filename + the full path?